freud/box.pyx

```diff
@@ -255,26 +259,26 @@ cdef class Box:
                 Fractional coordinate vectors: :math:`\left(3\right)` or :math:`\left(N, 3\right)`.
         """  # noqa: E501
         vecs = np.asarray(vecs)
-        if vecs.ndim > 2 or vecs.shape[vecs.ndim-1] != 3:
-            raise ValueError(
-                "Invalid dimensions for vecs given to makeFraction. "
-                "Valid input is an array of shape (3,) or (N,3).")
-
-        vecs = freud.common.convert_array(vecs)
-
-        if vecs.ndim == 1:
-            vecs[:] = self._makeFraction(vecs)
-        elif vecs.ndim == 2:
-            for i, vec in enumerate(vecs):
-                vecs[i] = self._makeFraction(vec)
-        return vecs
+        flatten = vecs.ndim == 1
+        vecs = np.atleast_2d(vecs)
+        vecs = freud.common.convert_array(vecs, shape=(None, 3))
+
+        vecs = self._makeFraction_many(vecs)
```

Comment:
Please refactor to remove the `_makeFraction` and `_makeFraction_many` methods, just put that logic directly in here. Same for the other methods.

Reply:
I concur, see my overall PR comment.